broker-util/oo-admin-repair

```diff
@@ -403,6 +418,37 @@ def find_unresponsive_servers(available_servers, auto_confirm)
   unresponsive_servers
 end
 
+def find_orphaned_envs(namespace=nil)
+  orphaned_envs = []
+  domains = []
+  if namespace
+    begin
+      domains = [Domain.find_by(:namespace => namespace)]
+    rescue Mongoid::Errors::DocumentNotFound
+      print_message "Unable to find domain. The domain #{namespace} does not exist. Skipping..."
+    rescue Exception => e
+      print_message "Unable to find domain #{namespace} with error: #{e.message}."
+    end
+    # Find all component_ids for every application in this domain
+    component_ids = domains[0].applications.map {|a| a.component_instances.map {|ci| ci._id}}.flatten
+  else
+    domains = Domain.all
+    # find all component_ids for every application
+    component_ids = Application.all.map {|a| a.component_instances.map {|ci| ci._id}}.flatten
```

Comment:
Alternatively, unless @a13m thinks they will run this to search the entirety of OpenShift Online for orphaned variables instead of simply repairing domains as they are reported you could just run a quick query to see how many domains or applications exist.  If it's determined to be "too big" then you could just print a warning telling the admin to run it against explicit domains.  That would be the lame way to approach this problem but it might the real world use cases right now.